src/Streamly/Internal/Data/Array/Foreign/Type.hs

```diff
@@ -223,7 +226,8 @@ fromPtr n ptr = unsafeInlineIO $ do
     fptr <- newForeignPtr_ ptr
     let end = ptr `plusPtr` n
     return $ Array
-        { aStart = fptr
+        { arrStart = ptr
+        , arrContents = undefined
```

Comment:
What happens when you `touch` `undefined`?
If we use `fromPtr`, will we be able to use the result in other combinators?